tools/cbmc/proofs/make_create_header_files.py

```diff
@@ -0,0 +1,144 @@
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import re
+
+CWD = os.getcwd()
+
+CREATE_HEADERS_FOR = ["lib/FreeRTOS/queue.c"]
+
+POSIX_INCLUDE = "lib/include/FreeRTOS_POSIX"
+TCP_INCLUDE = "lib/FreeRTOS-Plus-TCP"
+FreeRTOS_INCLUDE = "lib/include/private"
+
+
+FreeRTOS_HEAD = os.path.normpath(os.path.join(CWD, "..", "..", ".."))
+
+REGEX_START = re.compile(r"/\*CBMC_HEADER_START\:([\w\.]+)\*/")
+REGEX_END = re.compile(r"/\*CBMC_HEADER_END\*/")
+
+REGEX_INJECTED = re.compile(r"/\*CREATE_HEADER_FILES\*/")
+REGEX_INCLUDE = re.compile(r"#include\"([\w\.]+)\"")
+
+def get_absolute_header(header_name, file):
+    if "FreeRTOS_POSIX" in file:
+        absolut_header = os.path.normpath(
+            os.path.join(FreeRTOS_HEAD,
+                         *POSIX_INCLUDE.split("/"),
+                         header_name))
+    elif "FreeRTOS-Plus-TCP" in file:
+        absolut_header = os.path.normpath(
+            os.path.join(FreeRTOS_HEAD,
+                         *TCP_INCLUDE.split("/"),
+                         header_name))
+    else:
+        absolut_header = os.path.normpath(
+            os.path.join(FreeRTOS_HEAD,
+                         *FreeRTOS_INCLUDE.split("/"),
+                         header_name))
+    return absolut_header
+
+def create_header(file):
+    absolut_file = os.path.join(FreeRTOS_HEAD, *file.split("/"))
+    with open(absolut_file, "r") as source:
+        in_header = False
+        source_modified = ""
+        header_content = ""
+        header_name = None
+        for line in source:
+            start_matched = REGEX_START.match(line.replace(" ", ""))
+            end_matched = REGEX_END.match(line.replace(" ", ""))
+            if start_matched:
+                in_header = True
+                header_name = start_matched.group(1)
+                print(header_name)
+                source_modified += "/* CREATE_HEADER_FILES */\n"
+                source_modified += "#include \"{}\"\n".format(header_name)
+            if in_header:
+                header_content += line
+            else:
+                source_modified += line
+            if end_matched:
+                in_header = False
+    with open(absolut_file, "w") as source:
+        source.write(source_modified)
+    if header_name:
+        absolut_header = get_absolute_header(header_name, file)
+        print("Create:", absolut_header)
+        with open(absolut_header, "w") as out:
+            print("#ifndef QUEUE_DATASTRUCTURE_H", file=out)
+            print("#define QUEUE_DATASTRUCTURE_H", file=out)
+            out.write(header_content)
+            print("#endif /* QUEUE_DATASTRUCTURE_H */", file=out)
```

Comment:
Could this definition (`QUEUE_DATASTRUCTURE_H`) be obtained automatically from the file? I guess they could (from the start marker) and then it should work for any header in the list.